### PR TITLE
Spec test fast jit

### PR DESCRIPTION
**Pull request by TianlongLiang**
1. Remove unnecessary download llvm cache for classic/fast interpreter spec test
2. Add Fast JIT to spec case test

### CODE REVIEW

File: .github/workflows/compilation_on_android_ubuntu.yml [hunk 1]
```diff
@@ -355,21 +367,41 @@ jobs:
...
+          - running_mode: "classic-interp"
+            test_option: $SIMD_TEST_OPTIONS
+          - running_mode: "fast-interp"
+            test_option: $SIMD_TEST_OPTIONS
+          # aot and jit doesn't support multi module
+          - running_mode: "aot"
+            test_option: $MULTI_MODULES_TEST_OPTIONS
+          - running_mode: "jit"
+            test_option: $MULTI_MODULES_TEST_OPTIONS
+          # fast-jit only tested on default mode, exclude other three
```
Comment (wenyongh) [inline .github/workflows/compilation_on_android_ubuntu.yml:391]: `on tested` to `is only tested`?

File: .github/workflows/compilation_on_android_ubuntu.yml [hunk 2]
```diff
@@ -355,21 +367,41 @@ jobs:
...
+          - running_mode: "fast-jit"
+            test_option: $THREADS_TEST_OPTIONS
     steps:
       - name: checkout
         uses: actions/checkout@v3
 
       #only download llvm libraries in jit and aot mode
       - name: Get LLVM libraries
-        if: matrix.running_mode == 'jit' || matrix.running_mode == 'aot'
+        if: matrix.running_mode != 'classic-interp' && matrix.running_mode != 'fast-interp'
```
Comment (wenyongh) [inline .github/workflows/compilation_on_android_ubuntu.yml:404]: No need to change this line, no need to download llvm for fast jit mode

File: .github/workflows/compilation_on_android_ubuntu.yml [hunk 3]
```diff
@@ -382,7 +414,7 @@ jobs:
           key: ubuntu-20.04-${{ env.LLVM_CACHE_SUFFIX }}
 
       - name: Quit if cache miss
-        if: (matrix.running_mode == 'jit' || matrix.running_mode == 'aot') && steps.cache_llvm.outputs.cache-hit != 'true'
+        if: matrix.running_mode != 'classic-interp' && matrix.running_mode != 'fast-interp' && steps.cache_llvm.outputs.cache-hit != 'true'
```
Comment (wenyongh) [inline .github/workflows/compilation_on_android_ubuntu.yml:417]: Same as above

File: .github/workflows/compilation_on_android_ubuntu.yml [hunk 4]
```diff
@@ -394,7 +426,7 @@ jobs:
 
       #only install x32 support libraries when to run x86_32 cases
       - name: install x32 support libraries
-        if: matrix.test_option == '$DEFAULT_TEST_OPTIONS' || matrix.test_option == '$THREADS_TEST_OPTIONS'
+        if: (matrix.test_option == '$DEFAULT_TEST_OPTIONS' || matrix.test_option == '$THREADS_TEST_OPTIONS') && matrix.running_mode != 'fast-jit' && matrix.running_mode != 'jit'
```
Comment (wenyongh) [inline .github/workflows/compilation_on_android_ubuntu.yml:429]: The line is too long, can we put `&& matrix.running_mode != 'fast-jit' && matrix.running_mode != 'jit'` to the next line?

File: .github/workflows/compilation_on_android_ubuntu.yml [hunk 5]
```diff
@@ -404,6 +436,6 @@ jobs:
           sudo apt install -y g++-multilib lib32gcc-9-dev
 
       - name: run spec tests x86_32
-        if: matrix.test_option == '$DEFAULT_TEST_OPTIONS' || matrix.test_option == '$THREADS_TEST_OPTIONS'
+        if: (matrix.test_option == '$DEFAULT_TEST_OPTIONS' || matrix.test_option == '$THREADS_TEST_OPTIONS') && matrix.running_mode != 'fast-jit' && matrix.running_mode != 'jit'
```
Comment (wenyongh) [inline .github/workflows/compilation_on_android_ubuntu.yml:439]: Same as above

File: .github/workflows/compilation_on_sgx.yml [hunk 6]
```diff
@@ -322,11 +322,18 @@ jobs:
       matrix:
         running_mode: ["classic-interp", "fast-interp", "aot"]
         test_option: ["-x -p -s spec -P", "-x -p -s spec -S -P"]
+        exclude:
+          - running_mode: "classic-interp"
```
Comment (wenyongh) [inline .github/workflows/compilation_on_sgx.yml:326]: Had better add comments, here same as other file: `# classic-interp and fast-interp don't support simd`

File: .github/workflows/compilation_on_android_ubuntu.yml (hunk 2 above)
Comment (lum1n0us) [inline .github/workflows/compilation_on_android_ubuntu.yml:404]: maybe we shall add a step to prepare variables to control llvm library downloading and x86-32 test. In order to write those long `if` conditions fewer times.

**Review by wenyongh — COMMENTED** [review]
LGTM

**Pull request merged**